dialyzer.ignore-warnings

```diff
@@ -0,0 +1 @@
+Function deliver/1 only terminates with explicit exception
```

Comment:
This file is what we're ignoring in dialyzer. I fixed all the warnings I could, but this one comes from Bamboo. The issue is fixed on [`master`](https://github.com/thoughtbot/bamboo/pull/338) there, but upgrading to Bamboo 1.0 was kind of an involved process, so I figured I'd leave that for later and ignore this for now.